HeterogeneousCore/CUDAUtilities/interface/HostAllocator.h

```diff
@@ -28,7 +30,7 @@ namespace cms {
         using other = HostAllocator<U, FLAGS>;
       };
 
-      T* allocate(std::size_t n) const __attribute__((warn_unused_result)) __attribute__((malloc))
+      T* allocate CMS_THREAD_SAFE(std::size_t n) const __attribute__((warn_unused_result)) __attribute__((malloc))
```

Comment:
is it the right place, between `allocate` and `(std::size_t n)` ?

Reply:
That appears to be a legitimate place for C++ attributes, honestly I was following an example in
https://github.com/cms-sw/cmssw/blob/8ffadd15b51612959296f749c1f0269947d3b3f5/DataFormats/TrackReco/interface/Track.h#L74-L76

On a further thought I agree the placement looks a bit weird. @gartung, could you comment how picky the static analyzer is for the placement of `[[cms::thread_safe]]` attribute? Could it be placed e.g. before the function declaration? (that we seem to do with `CMS_SA_ALLOW` here
https://github.com/cms-sw/cmssw/blob/8ffadd15b51612959296f749c1f0269947d3b3f5/FWCore/Concurrency/interface/WaitingTaskHolder.h#L77
)

Reply:
as far as I know, both
```
CMS_THREAD_SAFE T* allocate(std::size_t n) const __attribute__((warn_unused_result)) __attribute__((malloc)) 
```
and
```
T* allocate(std::size_t n) const CMS_THREAD_SAFE __attribute__((warn_unused_result)) __attribute__((malloc)) 
```
(or after the other attributes) should be fine

Reply:
As I recall there was some trial and error on where to place the attribute to the tokenizer would process it and attach it to the correct statement.

Reply:
clang-check -ast-dump -ast-dump-filter=function-name will show you the AST the tokenizer produces and you can see if the attribute is attached to the function.